module/solutions/helper.js

```diff
@@ -1992,45 +2011,113 @@ module.exports = class SolutionsHelper {
 						message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
 					})
 				}
+				// Build the $addToSet updateObject
+				let updateObject = { $addToSet: {} }
+				let validationExcludedEntitiesKeys = []
+				if (
+					userDetails.userInformation.roles.includes(CONSTANTS.common.ADMIN_ROLE) ||
+					userDetails.userInformation.roles.includes(CONSTANTS.common.TENANT_ADMIN)
+				) {
+					// Fetch tenant details to validate organization codes
+					let tenantDetails = await userService.fetchTenantDetails(tenantId, userDetails.userToken)
```

Comment:
@MallanagoudaB we have to throw error if user service call fails. please handle error case